subsys/bluetooth/host/audio/mpl.c

```diff
@@ -2204,6 +2247,77 @@ void (*command_handlers[BT_MCS_MEDIA_STATE_LAST])(struct mpl_cmd command,
 	seeking_state_command_handler
 };
 
+#ifdef CONFIG_BT_OTS
+/* Find a track by ID
+ *
+ * If found, return pointers to the group of the track and the track
+ * otherwise, the pointers returned are NULL
+ */
+static void find_track_by_id(struct mpl_mediaplayer_t *pl, uint64_t id,
+			     struct mpl_group_t **group, struct mpl_track_t **track)
+{
+	struct mpl_group_t *tmp_group = pl->group;
+	struct mpl_track_t *tmp_track;
+
+	while (tmp_group->prev != NULL) {
+		tmp_group = tmp_group->prev;
+	}
```

Comment:
This is to find the head, right? 

Could the head be stored in the `struct mpl_mediaplayer_t *pl` for direct and fast access?

Reply:
Yes.  But I consider that out of scope for this PR.